components/hidden_data_box/commons/hidden_data_box.lua

```diff
@@ -107,6 +152,8 @@ function HiddenDataBox.validateTier(tierString, tierMode)
 		)
 	end
 
+	-- For types we want to return the normalized value
+	-- For tiers we want to return the input
 	tierValue = tierMode == TIER_MODE_TYPES and tierValue or tierString
 
 	return tierValue, warning
```

Comment:
Shouldn't the exact same code exist in `Infobox/League`, and if so, is there an easy way to share it?

Reply:
The code in League is somewhat different. In league we do two things with Tier.

* We grab the input (a number) and transform it into a display text (& link). Here we check if the tier is valid
* We take the input raw into LPDB (and wiki variable)

We also do two thing with TierType

* Grab the input (a string) and looks up the normalized name. Here we check if the tiertype is valid
* Stores the normalized name in LPDB and wiki variable

In HBD we only do the latter step, which has no validation in League infobox. Hence they vary quite a lot. The step we do in League for the equivalent to this function is 
 ```lua
 	Variables.varDefine('tournament_liquipediatier', args.liquipediatier)
	Variables.varDefine(
		'tournament_liquipediatiertype',
		Tier.text.types
			and Tier.text.types[string.lower(args.liquipediatiertype or '')]
			or args.liquipediatiertype
	)
	 ```